src/services/collections.js

```diff
@@ -1,19 +1,54 @@
-import config from 'config';
+import { URL } from 'url';
 
 import fetch from '../utils/fetch.js';
+import logger from '../utils/logger.js';
 
-export const fetchCollections = async () => {
-  const collections = await fetch(config.get('collectionsUrl'));
+export const fetchCollections = async collectionsConfig => {
+  const unqiueCollections = new Map();
 
-  return Object.keys(collections).reduce((result, collectionName) => {
-    if (collections[collectionName].endpoint) {
-      result.push({
-        name: collectionName,
-        id: collections[collectionName].id,
-        endpoint: collections[collectionName].endpoint,
-      });
+  const result = await Promise.allSettled(collectionsConfig.map(async item => {
+    let collections = [];
+
+    if (typeof item === 'string') {
+      collections = await fetch(item);
```

Comment:
As I understand it, we don't want to `await` in the loop if we do `allSettled`, we want to return the promise.

Reply:
Actually, we return a promise by defining the function as `async`. The `await` inside this function ensures sequential execution within the function, while the function itself remains asynchronous. So, I can confirm that all fetch operations are indeed executed in parallel.